### PR TITLE
Session handling [Selected columns remain same until admin not logged…

### PR DESCRIPTION
**Pull request by BhagyashreeBhoyar**
… out].

Maintain admin login session till admin logout.

### CODE REVIEW

File: src/components/DataGrid.jsx [hunk 1]
```diff
@@ -234,11 +297,24 @@ class DataGrid1 extends Component {
     this.setState({advanceFilterIsOpen: false});
   }
   setValuesOfVisibleColumnConfig(values, selectValue){
```
Comment (pepcus-pratik) [inline src/components/DataGrid.jsx]: @BhagyashreeBhoyar , please mention the purpose of this method

File: src/components/SplashPage.jsx [hunk 2]
```diff
@@ -107,6 +107,8 @@ class SplashPage extends Component {
     })
   }
   checkAdminCredential(){
+     let data = sessionStorage.getItem('isAdminLogin');
```
Comment (pepcus-pratik) [inline src/components/SplashPage.jsx]:
```suggestion
     let isAdminLogin = sessionStorage.getItem('isAdminLogin');
```

File: src/components/SplashPagePrePopulated.jsx [hunk 3]
```diff
@@ -100,30 +100,37 @@ class SplashPage extends Component {
...
-    } = this.props;
-    if(this.state.message) {
-      if (id !== adminId || password !== adminPassword) {
-        return (
-          <div className={'errorPopupContainer'}>
-            <h5>{invalidAdminMsg}</h5>
-          </div>
-        );
+  checkAdminCredential() {
+    let data = sessionStorage.getItem('isAdminLogin');
```
Comment (pepcus-pratik) [inline src/components/SplashPagePrePopulated.jsx]:
```suggestion
    let isAdminLogin = sessionStorage.getItem('isAdminLogin');
```

File: src/components/DataGrid.jsx [hunk 4]
```diff
@@ -160,13 +162,13 @@ class DataGrid1 extends Component {
...
       selectedStudentsCheck: [],
-      selectValue: true,
+      selectValue: this.getSelectValue(),
       students:[],
       metaData: gridHeaderData(),
       columnOptionIsOpen:false,
       isStudentDataSet: false,
       advanceFilterIsOpen: false,
-      visibleColumnConfig: {
+      visibleColumnConfig: this.getVisibleColumnConfigObject(),/* {
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/DataGrid.jsx]: I removed this commented code.

File: src/components/DataGrid.jsx [hunk 5]
```diff
@@ -234,11 +297,24 @@ class DataGrid1 extends Component {
     this.setState({advanceFilterIsOpen: false});
   }
   setValuesOfVisibleColumnConfig(values, selectValue){
+    let count = 0;
+    for(let k in values) {
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/DataGrid.jsx]: ok, I replaced k by key.

File: src/components/DataGrid.jsx (hunk 1 above)
Comment (BhagyashreeBhoyar) [reply src/components/DataGrid.jsx]: It stores the selected column in  redux store.

File: src/components/SplashPage.jsx (hunk 2 above)
Comment (BhagyashreeBhoyar) [reply src/components/SplashPage.jsx]: Done.

File: src/components/SplashPagePrePopulated.jsx (hunk 3 above)
Comment (BhagyashreeBhoyar) [reply src/components/SplashPagePrePopulated.jsx]: Done.

File: src/components/DataGrid.jsx [hunk 6]
```diff
@@ -6,26 +6,28 @@ import { Redirect, Link } from 'react-router-dom';
...
+  setRedirectValueAction,
+  setVisibleColumnConfigAction,
 } from  '../actions/studentRegistrationActions';
 import {
   stateOfRedirect,
   stateOfAdminLogin,
 } from '../reducers/studentRegistrationReducer';
 import AdvanceSearch from './AdvanceSearch';
-import IdCardWrapper from './IdCardWrapper';
+import IdCardButtons from './IdCardButtons';
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/DataGrid.jsx]: Done.

File: src/components/DataGrid.jsx [hunk 7]
```diff
@@ -6,26 +6,28 @@ import { Redirect, Link } from 'react-router-dom';
...
 import {
   yjsgHeader,
   goBackBtnText,
   adminPassword,
 } from '../utils/yjsgConstants';
+import PropTypes from 'prop-types';
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/DataGrid.jsx]: Done.

File: src/components/DataGrid.jsx [hunk 8]
```diff
@@ -204,7 +206,44 @@ class DataGrid1 extends Component {
...
+    let DefaultSelectValue = true;
+    if(this.props.selectValue !== undefined) {
+      return this.props.selectValue;
+    }
+    else {
+      return DefaultSelectValue;
+    }
+  }
+  getVisibleColumnConfigObject() {
+    let OriginalVisibleColumnConfigObject = {
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/DataGrid.jsx]: Done.

**Pull request merged**